cloud_images/centos7/dcos_vol_setup.sh

```diff
@@ -53,7 +53,12 @@ function main {
     echo "Adding entry to fstab: $fstab"
     echo "$fstab" >> /etc/fstab
     echo -n "Mounting: $device to $mount_location"
-    mount "$mount_location"
+    until grep ^$device /etc/mtab > /dev/null; do sleep 1; echo -n .; mount "$mount_location" || :; done
+    if [ "$mount_location" = "/var/log" ]; then
+      echo "Recreating /var/log/journal"
+      mkdir /var/log/journal
+      systemd-tmpfiles --create --prefix /var/log/journal
+    fi
```

Comment:
I previously did tests with `Before=` and any non-existing units are being ignored. Remember though that dcos_vol_setup.service is being executed at every bootup, not just on first boot. In most cases the partitions will exist after first boot and most of the code won't ever be executed again. But it's perceivable that some users will destroy an EBS and reattach it to a running cluster. In that case the dcos_vol_setup would be run again to initialize the partition and dcos-setup.service would exist.

Re: sending output somewhere else. I experimented with a simple subshell that accepts the data on stdin and ignores output errors. E.g.
```
bash -c "set -e; trap '' PIPE; echo foo | (cat || :); sleep 3" | :
```
Here the echo output is being piped into the subshell. The cat get's interrupted but the signal is being ignored and because the failed cat results in a noop the script won't get aborted and the following sleep is executed.